src/App.tsx

```diff
@@ -8,37 +8,87 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getUsers } from './api/api';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasPostsError, setHasPostsError] = useState(false);
+  const [hasUsersError, setHasUsersError] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  useEffect(() => {
+    getUsers()
+      .then(setUsers)
+      .catch(() => setHasUsersError(true))
+      .finally(() => setHasUsersError(false));
+  }, []);
+
+  const noPostsWarn = selectedUser && !posts.length
+    && !isLoading && !hasPostsError;
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                  setPosts={setPosts}
+                  setIsLoading={setIsLoading}
+                  setHasPostsError={setHasPostsError}
+                  setSelectedPost={setSelectedPost}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
 
-                <Loader />
+                {isLoading && <Loader />}
+
+                {!selectedUser && !hasUsersError && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
+
+                {hasUsersError && (
+                  <div
+                    className="notification is-danger"
+                  >
```

Comment:
It would be more readable
```suggestion
                  <div className="notification is-danger">
```